src/controllers/hooks/router-store/index.tsx

```diff
@@ -18,3 +26,32 @@ export const useRouterActions = (): RouterActionsType => {
 
   return allActions;
 };
+
+export const useLocation = (): Location => {
+  const [routerState] = useRouterStore();
+
+  return routerState.location;
+};
+
+export const useHistory = (): BrowserHistory => {
```

Comment:
We always said that passing history around was a bad idea. Are we changing idea now?

Reply:
Yeah I agree here, I think we should be careful about exposing it